### PR TITLE
Add the capability to know which CVE’s affect an agent - Implementation

### PR DESCRIPTION
**Pull request by DProvinciani**
|Related issue|
|---|
| https://github.com/wazuh/wazuh/issues/7479 |

## Description

This pull request includes all the changes implemented as part of the epic: [Add the capability to know which CVE’s affect an agent](https://github.com/wazuh/wazuh/issues/7479). For details about the requirements, check the **description** section of the issue.

### Testing done

- We implemented unit tests to cover 100% of the new functions created and keep the coverage level of the ones that already existed. This was done as part of the issues: https://github.com/wazuh/wazuh/issues/7514, https://github.com/wazuh/wazuh/issues/7516, and https://github.com/wazuh/wazuh/issues/7518.
- We implemented integration tests to cover all the new commands included in Wazuh DB to manage the new database table that holds de vulnerabilities information. This was done as part of the issue https://github.com/wazuh/wazuh-qa/issues/1069.
- We executed full exploratory testing that allowed us to find some issues that were properly fixed. This was done as part of the issue https://github.com/wazuh/wazuh/issues/7519.
- We executed static analysis over the implementation by using both `scan-build` and `Coverity`. For details about the issues and fixes check https://github.com/wazuh/wazuh/issues/7611.

## Tests

- Compilation without warnings in every supported platform
  - [x] Linux
- [x] Source installation
- [x] Package installation
- [x] Source upgrade
- [x] Package upgrade

- Memory tests for Linux
  - [x] Scan-build report
  - [x] Coverity
  - [x] Valgrind (memcheck and descriptor leaks check)

### CODE REVIEW

**Review by chemamartinez — DISMISSED** [review]

File: src/wazuh_db/schema_agents.sql [hunk 1]
```diff
@@ -340,9 +340,19 @@ CREATE TABLE IF NOT EXISTS sync_info (
...
+    cve TEXT,
+    PRIMARY KEY (name, version, architecture, cve)
+);
+CREATE INDEX IF NOT EXISTS packages_id ON vuln_cves (name);
+CREATE INDEX IF NOT EXISTS cves_id ON vuln_cves (cve);
+
 BEGIN;
 
-INSERT INTO metadata (key, value) VALUES ('db_version', '7');
+INSERT INTO metadata (key, value) VALUES ('db_version', '8');
```
Comment (vikman90) [inline src/wazuh_db/schema_agents.sql:355]: Version 4.2 already upgraded to DB version 7. We should stay at 7.

File: src/wazuh_db/schema_upgrade_v8.sql [hunk 2]
```diff
@@ -0,0 +1,22 @@
+/*
+ * SQL Schema for upgrading databases
+ * Copyright (C) 2015-2020, Wazuh Inc.
```
Comment (vikman90) [inline src/wazuh_db/schema_upgrade_v8.sql:3]: We are un 2021!
However, please delete this file.

**Review by vikman90 — CHANGES_REQUESTED** [review]

File: src/wazuh_db/schema_agents.sql (hunk 1 above)
Comment (DProvinciani) [reply src/wazuh_db/schema_agents.sql]: Fixed

File: src/wazuh_db/schema_upgrade_v8.sql (hunk 2 above)
Comment (DProvinciani) [reply src/wazuh_db/schema_upgrade_v8.sql]: Fixed

**Review by vikman90 — APPROVED** [review]

**Pull request merged**